apps/backend/src/modules/users/user.repository.ts

```diff
@@ -148,14 +164,20 @@ class UserRepository implements Repository {
 			.where({ email })
 			.first()
 			.withGraphFetched(
-				`[${RelationName.USER_DETAILS}.[${RelationName.AVATAR}], ${RelationName.USER_TASK_DAYS}, ${RelationName.ONBOARDING_USER_ANSWERS}, ${RelationName.QUIZ_USER_ANSWERS}]`,
+				`[${RelationName.USER_DETAILS}.[${RelationName.AVATAR}], ${RelationName.USER_TASK_DAYS}, ${RelationName.ONBOARDING_USER_ANSWERS}, ${RelationName.QUIZ_USER_ANSWERS},${RelationName.USER_TASKS}]`,
 			)
 			.castTo<undefined | UserWithAvatarFile>();
 
 		return user
 			? UserEntity.initialize({
 					avatarFileId: user.userDetails.avatarFileId,
 					avatarUrl: user.userDetails.avatarFile?.url ?? null,
+					completionTasksPercentage:
+						(user.userTasks.filter(
+							(task) => task.status === TaskStatus.COMPLETED,
+						).length /
+							user.userTasks.length) *
```

Comment:
Let's create a helper for that instead. We can use this helper in repository then